starwars/src/App.js

```diff
@@ -22,17 +28,46 @@ class App extends Component {
         return res.json();
       })
       .then(data => {
-        this.setState({ starwarsChars: data.results });
+        this.setState({
+          starwarsChars: data.results,
+          previousPage: data.previous,
+          nextPage: data.next
+        });
       })
       .catch(err => {
         throw new Error(err);
       });
   };
 
+  nextPage = () => {
+    if (this.state.nextPage === null) {
+      this.getCharacters(this.state.currentPage);
```

Comment:
In production, we have to start to consider how long it's taking these operations to run. Especially when we use create-react-app because it's huge and takes a minute to load to begin with. I'm not an expert on this, but one of the easiest ways we can help make the user experience smooth and fast is to not make extraneous API calls. When you are on the first or last page and you run this.getCharacters it has to make an api call to go get the data you already have, then React does a re-render when you run this.setState . I would 
```js
nextPage = () => {
    if (this.state.nextPage !== null) {
      this.getCharacters(this.state.nextPage);
    }
  };
```
instead. It's ok that it doesn't do anything if this.state.nextPage is equal to null